borg/hashindex.pyx

```diff
@@ -175,9 +192,39 @@ cdef class ChunkIndex(IndexBase):
 
     def __contains__(self, key):
         assert len(key) == self.key_size
-        data = <int *>hashindex_get(self.index, <char *>key)
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
         return data != NULL
 
+    def incref(self, key):
+        """Increase refcount for 'key', return (refcount, size, csize)"""
+        assert len(key) == self.key_size
+        data = <int32_t *>hashindex_get(self.index, <char *>key)
+        if not data:
+            raise KeyError(key)
+        cdef int32_t refcount = _le32toh(data[0])
+        if refcount != INT32_MAX:
+            if INT32_MAX - refcount < 1:
```

Comment:
*If* we consider this case (and as I wrote above it might be valid to ignore this), then we'd need to do something like

```py
# refcount = -something
if refcount < 0:
  refcount = INT32_MAX
elif refcount != INT32_MAX:
  refcount += / -= 1
```

For _add we'd need to check sign of operands and choose appropiate check from there. Overflowing is UB and can't be used.

> (or just plain cache corruption of some other sort).

There are no integrity checks for the indices, but they should be added.